src/yaml/mapping.cr

```diff
@@ -87,12 +91,24 @@ module YAML
       {% end %}
     {% end %}
 
+    {% if extra %}
+      @{{extra.id}} : Hash(String, ::YAML::Any)
+
+      def {{extra.id}}
+        @{{extra.id}}
+      end
+    {% end %}
+
     def initialize(%pull : ::YAML::PullParser)
       {% for key, value in properties %}
         %var{key.id} = nil
         %found{key.id} = false
       {% end %}
 
+      {% if extra %}
+        {{extra.id}} = Hash(String, ::YAML::Any).new
```

Comment:
Why not use directly `@{{extra.id}}` (and remove the `@{{extra.id}} = {{extra.id}}` at the end of the macro) ?